servant-server/src/Servant/Server/Internal/RoutingApplication.hs

```diff
@@ -51,8 +51,18 @@ data RouteMismatch =
   | WrongMethod        -- ^ a more informative "you just got the HTTP method wrong" error
   | UnsupportedMediaType -- ^ request body has unsupported media type
   | InvalidBody String -- ^ an even more informative "your json request body wasn't valid" error
-  | HttpError Status (Maybe BL.ByteString)  -- ^ an even even more informative arbitrary HTTP response code error.
-  deriving (Eq, Ord, Show)
+  | HttpError Status [Header] (Maybe BL.ByteString)  -- ^ an even even more informative arbitrary HTTP response code error.
+  | RouteMismatch Response -- ^ an arbitrary mismatch with custom Response.
```

Comment:
The `ResponseRaw` constructor makes `Eq` impossible for `Response`. See: https://hackage.haskell.org/package/wai-3.0.3.0/docs/src/Network-Wai-Internal.html#Response